src/baseline.rs

```diff
@@ -340,3 +356,82 @@ fn need_retry(res: Result<(), crates_index::Error>) -> anyhow::Result<bool> {
         Err(err) => Err(err.into()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crates_index::{Crate, Version};
+
+    use super::choose_baseline_version;
+
+    fn new_mock_version(version_name: &str, yanked: bool) -> Version {
+        // `crates_index::Version` cannot be created explicitly, as all its fields
+        // are private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({
+            "name": "test-crate",
+            "vers": version_name,
+            "deps": [],
+            "features": {},
+            "yanked": yanked,
+            "cksum": "00".repeat(32),
+        }))
+        .unwrap()
+    }
+
+    fn new_crate(versions: &[Version]) -> Crate {
+        // `crates_index::Crate` cannot be created explicitly, as its field
+        // is private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({ "versions": versions })).unwrap()
```

Comment:
From what I remember, we're trying to use `expect` (or any alternatives with custom error messages) instead of `unwrap` for better debugging messages.

Reply:
Changed all `unwrap`s to `expect`s, please have a look at the messages.